tests/src/test_cases/test_led.c

```diff
@@ -0,0 +1,129 @@
+#if LV_BUILD_TEST
+#include "../lvgl.h"
+
+#include "unity/unity.h"
+
+static lv_obj_t * active_screen = NULL;
+static lv_event_code_t exp_evt_code;
+static bool event_triggered = false;
+
+void setUp(void)
+{
+    active_screen = lv_scr_act();
+}
+
+void tearDown(void)
+{
+    lv_obj_clean(active_screen);
+}
+
+void test_led_render_1(void)
+{
+    lv_obj_t * led  = lv_led_create(active_screen);
+    lv_obj_align(led, LV_ALIGN_CENTER, -80, 0);
+    lv_led_off(led);
+
+    led  = lv_led_create(active_screen);
+    lv_obj_align(led, LV_ALIGN_CENTER, 0, 0);
+    lv_led_set_brightness(led, 150);
+    lv_led_set_color(led, lv_palette_main(LV_PALETTE_RED));
+
+    led  = lv_led_create(active_screen);
+    lv_obj_align(led, LV_ALIGN_CENTER, 80, 0);
+    lv_led_on(led);
+
+    /* TODO: Add screen compare assert. */
+}
+
+void test_led_set_color_works(void)
+{
+    lv_obj_t * led  = lv_led_create(active_screen);
+    lv_led_t * ledObj = (lv_led_t *)led;
+
+    /* Test by changing colors randomly. */
+    lv_led_set_color(led, lv_palette_main(LV_PALETTE_AMBER));
+    TEST_ASSERT_EQUAL(lv_palette_main(LV_PALETTE_AMBER).full, ledObj->color.full);
+
+    lv_led_set_color(led, lv_palette_main(LV_PALETTE_GREY));
+    TEST_ASSERT_EQUAL(lv_palette_main(LV_PALETTE_GREY).full, ledObj->color.full);
+
+    lv_led_set_color(led, lv_palette_main(LV_PALETTE_RED));
+    TEST_ASSERT_EQUAL(lv_palette_main(LV_PALETTE_RED).full, ledObj->color.full);
```

Comment:
> This means we are only testing the public API surface and its apparent behavior to the user, which is a better way of testing in my opinion.

I agree with this. Actually there should be an `lv_led_get_color` function too which could be used here. 

Similarly, instead of `ledObj->bright`, `lv_led_get_brightness(led)` should be used.

Reply:
It would still be ideal to use a screenshot in addition to just checking that the setters and getters work.

Reply:
@embeddedt I will be adding a screenshot in `test_led_render_1()` once @kisvegabor uploads it here.

Reply:
Here it is:
![image](https://user-images.githubusercontent.com/7599318/190363423-664564c7-3a13-4539-8169-820731bb58b8.png)